app/assets/javascripts/components/nav.jsx

```diff
@@ -0,0 +1,220 @@
+  },
+
+  showSettings(event) {
+    event.preventDefault();
+  },
+
+  render() {
+    let navBar;
+    let navClass;
+    let myDashboard;
+    let forAdmin;
+    let disableTraining;
+    let help;
+    let Sandbox;
+    let wikiEd;
+    let languageSwitcherEnabled;
+    let loggingLinks;
+    if (this.state.languageSwitcherEnabled)
+    {
+      languageSwitcherEnabled = (
+        <li>
+          <button className="uls-trigger">
+            {I18n.locale}
+          </button>
+        </li>
+      );
+    }
+    if (this.state.usersignedin)
+    {
+      loggingLinks = (
+        <span>
+          <li>
+            <b><a href={this.state.rootUrl} className="current-user">{this.state.currentUser}</a></b>
+          </li>
+          <li>
+            <a href={this.state.destroyUrl} className="current-user">{I18n.t('application.log_out')}</a>
+          </li>
+        </span>
+      );
+    } else {
+      loggingLinks = (
+        <li>
+          <a href={this.state.omniauthUrl}>
+            <i className="icon icon-wiki-logo"></i>
+            {I18n.t('application.log_in')}
+            <span className="expand">
+              {I18n.t('application.sign_up_log_in_extended')}
+            </span>
+          </a>
+        </li>
+      );
+    }
+    if (this.state.usersignedin === true) {
+      myDashboard = (
+        <li>
+          <Link to={this.state.rootUrl} activeClassName="active">{I18n.t('application.my_dashboard')}</Link>
+        </li>
+      );
+    }
+    if (this.state.ifadmin === true) {
+      forAdmin = (
+        <li>
+          <Link to="/admin" activeClassName="active">Admin</Link>
+        </li>
+      );
+    }
+    if (this.state.disableTraining === false) {
+      disableTraining = (
+        <li>
+          <Link to={this.state.trainingurl} activeClassName="active">{I18n.t('application.training')}</Link>
+        </li>
+      );
+    }
+    if ((this.state.usersignedin || this.state.helpDisabled) === false) {
+      help = (
+        <li>
+          <Link to={this.state.askUrl} activeClassName="active">{I18n.t('application.help')}</Link>
+        </li>
+      );
+    }
+    if (this.state.userPermissions) {
+      Sandbox = (
+        <li>
+          <Link to="https://en.wikipedia.org/wiki/Special:MyPage/sandbox" activeClassName="active">My Sandbox</Link>
+        </li>
+      );
+    }
+    if (!this.state.wikiEd) {
+      wikiEd = (
+        <span id="span_wikied">
+          <li>
+            <Link to="https://meta.wikimedia.org/wiki/Programs_%26_Events_Dashboard" activeClassName="active">{I18n.t('application.documentation')}</Link>
+          </li>
+          <li>
+            <Link to="/feedback" activeClassName="active">{I18n.t('application.report_problem')}</Link>
+          </li>
+        </span>
+      );
+    }
+    if (this.state.fluid)
+    {
+      navClass = "top-nav fluid";
+    } else {
+      navClass = "top-nav";
+    }
+    if (this.state.width < 500)
```

Comment:
Maybe we should factor these if/else branches out into two separate presentational components, one for mobile and one for desktop?